src/app/containers/TestExamWrapper/index.tsx

```diff
@@ -0,0 +1,263 @@
+import { QuestionManage } from "app/components/QuestionManage";
+import { useUpdateUserAnswer } from "mutations/userAnswer";
+import { ModalCustomization } from "app/components/ModalCustomization";
+import { useSubmitTheExam } from "mutations/userExam";
+
+const cx = classNames.bind(styles);
+
+export const TestExamWrapper = () => {
+  const navigate = useNavigate();
+
+  const { mutateAsync: mutateAsyncUpdateUserAnswer } = useUpdateUserAnswer();
+
+  const { userExamId } = useParams();
+  const { data: userExamByOwner } = useGetUserExamByOwner(userExamId as string);
+
+  const examId = userExamByOwner?.original_exam;
+  const { data: questionsByExamId } = useGetQuestionsByExamId(examId as string);
+
+  const { mutateAsync: mutateAsyncSubmitTheExam } = useSubmitTheExam();
+
+  const [score, setScore] = useState(-1);
+  const [userExamStatus, setUserExamStatus] = useState(false);
+
+  const [arrUserAnswer, setArrUserAnswer] = useState(
+    userExamByOwner &&
+      userExamByOwner?.user_answers[0] &&
+      userExamByOwner.user_answers[0].answers_id.length > 0
+      ? userExamByOwner.user_answers[0].answers_id
+      : [],
+  );
+
+  const [lastUpdatedPosition, setLastUpdatedPosition] = useState<number | null>(
+    null,
+  );
+  const countUserAnswerDone = (arr?: Array<string>) => {
+    let count = 0;
+    const array = arr ? arr : [];
+    for (const element of array) {
+      if (element !== "") {
+        count++;
+      }
+    }
+    return count;
+  };
+
+  const [numberAnswerDone, setNumberAnswerDone] = useState(
+    countUserAnswerDone(arrUserAnswer),
+  );
+  useEffect(() => {
+    if (
+      userExamByOwner?.user_answers &&
+      userExamByOwner.user_answers.length > 0
+    ) {
+      setArrUserAnswer(userExamByOwner.user_answers[0].answers_id);
+      setNumberAnswerDone(
+        countUserAnswerDone(userExamByOwner.user_answers[0].answers_id),
+      );
+    }
+    if (userExamByOwner?.is_completed) {
+      setScore(userExamByOwner?.score!);
+    } else {
+      setScore(-1);
+    }
+    setUserExamStatus(userExamByOwner?.is_completed!);
+  }, [userExamByOwner]);
+  const [openPopupSubmit, setOpenPopupSubmit] = useState(false);
+  const handleClosePopup = useCallback(() => {
+    setOpenPopupSubmit(false);
+  }, [openPopupSubmit]);
+
+  const handleOpenPopup = useCallback(() => {
+    setOpenPopupSubmit(true);
+  }, [openPopupSubmit]);
+
+  const changeAnswer = async (position: number, value: string) => {
+    try {
+      if (arrUserAnswer) {
+        arrUserAnswer[position] = value;
+
+        setArrUserAnswer(arrUserAnswer);
+      }
+
+      setNumberAnswerDone(countUserAnswerDone(arrUserAnswer));
+
+      setLastUpdatedPosition(position);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+  useEffect(() => {
+    const updateAnswerId = arrUserAnswer
+      ? arrUserAnswer[lastUpdatedPosition || 0]
+      : "";
+
+    if (updateAnswerId !== "") {
+      mutateAsyncUpdateUserAnswer({
```

Comment:
why put mutateAsyncUpdateUserAnswer in useEffect

Reply:
I thought wrong about result of api update user answer so I used useEffect hook,  i improved this code block by useCallback,